### ISSUE TITLE
Schema validation error thrown during reset token expiration

### ISSUE THREAD

**Issue by serkandurusoy** [meteor/meteor#7734]
As carried over from #6462 per request from @tmeasday 

The task that runs for password reset token expirations cause the following error to be thrown from collection2:

```
Exception in setInterval callback: Error: services.password.reset.$or is not allowed by the schema
     at getErrorObject (packages/aldeed_collection2-core/lib/collection2.js:451:1)
     at [object Object].doValidate (packages/aldeed_collection2-core/lib/collection2.js:433:1)
     at [object Object].Mongo.Collection.(anonymous function) [as update] (packages/aldeed_collection2-core/lib/collection2.js:185:1)
     at AccountsServer.Ap._expirePasswordResetTokens (packages/accounts-base/accounts_server.js:1137:14)
     at packages/accounts-base/accounts_server.js:1174:14
     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
     at packages/meteor/timers.js:6:1
     at runWithEnvironment (packages/meteor/dynamics_nodejs.js:110:1)
```

**Comment by tmeasday** [comment]
This seems like a bug in SS dealing with `$push: { "field": { or: ... } }`

**Comment by serkandurusoy** [comment]
Hmm, I'll cross post over there as well

**Comment by tarunbatra** [comment]
i'm getting the
`Exception in setInterval callback: MongoError: Cannot apply $pull to a non-array value`
error as others have mentioned in #6462 . Any updates on this?

**Comment by tmeasday** [comment]
@tarunbatra What Meteor version are you on? Which version of `npm-mongo` are you using?

**Comment by tarunbatra** [comment]
@tmeasday 
Meteor: `1.4.1.1`
npm-mongo : `1.5.48`

**Comment by tmeasday** [comment]
@tarunbatra apologies, I mean to ask your `accounts-base` version.

**Comment by tmeasday** [comment]
(if it's not `1.2.11` I suggest updating it)

**Comment by tarunbatra** [comment]
@tmeasday yep that was it. Thanx! :+1:

**Comment by abernix** [comment]
SimpleSchema shouldn't really be throwing an error for this, but the code that is causing this does not match a Mongo spec.  It just happens to work currently because [the actual value side of the `$unset` field doesn't matter](https://docs.mongodb.com/manual/reference/operator/update/unset/#up._S_unset).

While this update clause would have been correct with the original `$pull` methodology, the original implementation was found to have a bug [as commented here](https://github.com/meteor/meteor/pull/7534#discussion-diff-74602906) and it was changed to use `$unset`.  However, an `$or` on the value-side of an `$unset` field element wouldn't do anything.

Anyhow, this issue was also found by @mitar in https://github.com/meteor/meteor/pull/7817#discussion_r80365772 and will be fixed by that PR.  If it doesn't land soon, we can make a separate PR to resolve this.  I will leave this open until that happens.

**Comment by abernix** [comment]
Thank you for reporting this!  This should have been fixed by #7817 in Meteor 1.4.2 and higher.  Please report back if this problem still affects you after updating to Meteor 1.4.2.x.
